cmd/minikube/cmd/start.go

```diff
@@ -375,11 +375,18 @@ func runStart(cmd *cobra.Command, args []string) {
 	if driverName == driver.None {
 		prepareNone()
 	}
-	if viper.GetBool(waitUntilHealthy) {
-		if err := bs.WaitCluster(config.KubernetesConfig, viper.GetDuration(waitTimeout)); err != nil {
-			exit.WithError("Wait failed", err)
+
+	var podsToWaitFor map[string]struct{}
+
+	if !viper.GetBool(waitUntilHealthy) {
+		// only wait for apiserver if wait=false
+		podsToWaitFor = map[string]struct{}{
+			"apiserver": struct{}{},
 		}
 	}
+	if err := bs.WaitCluster(config.KubernetesConfig, viper.GetDuration(waitTimeout), podsToWaitFor); err != nil {
```

Comment:
Do you mind making this into a slice of strings instead of a map of struct? The map is useful internally, but it doesn't feel particularly friendly to clients, and not more intuitive than a slice of strings.

Some ideas to make the interface clearer.

- Rename `WaitCluster` to `WaitForPods` 
- Accept nil slice (as you do now) or a constant []string{ALL_PODS} to wait for all pods
- Accept an empty slice ([]string{}) to wait for nothing (as apposed to a nil slice)

Reply:
For sure, added all of those in.